routes/validators.go

```diff
@@ -354,6 +356,127 @@ func (fes *APIServer) GetValidatorByPublicKeyBase58Check(ww http.ResponseWriter,
 	}
 }
 
+// GetCurrentEpochProgressResponse encodes the current epoch entry, the leader schedule for it, and the
+// progress throughout the epoch. Based on the data returned, the client can determine the chain's
+// progress through the epoch, the current leader and all upcoming leaders.
+type GetEpochProgressResponse struct {
+	// The full epoch entry object
+	EpochEntry     lib.EpochEntry  `safeForLogging:"true"`
+	LeaderSchedule []UserInfoBasic `safeForLogging:"true"`
+
+	CurrentView      uint64 `safeForLogging:"true"`
+	CurrentTipHeight uint64 `safeForLogging:"true"`
+}
+
+type UserInfoBasic struct {
+	PublicKeyBase58Check string `safeForLogging:"true"`
+	Username             string `safeForLogging:"true"`
+}
+
+func (fes *APIServer) GetCurrentEpochProgress(ww http.ResponseWriter, req *http.Request) {
+	// Fetch the current snapshot from the blockchain. We use the latest uncommitted tip.
+	utxoView, err := fes.backendServer.GetBlockchain().GetUncommittedTipView()
+	if err != nil {
+		_AddInternalServerError(ww, "GetCurrentEpochProgress: problem fetching uncommitted tip")
+		return
+	}
+
+	// Get the current epoch number.
+	currentEpochEntry, err := utxoView.GetCurrentEpochEntry()
+	if err != nil {
+		_AddInternalServerError(ww, "GetCurrentEpochProgress: problem fetching current epoch number")
+		return
+	}
+
+	// Get the current uncommitted tip.
+	currentTip := fes.backendServer.GetBlockchain().BlockTip()
+
+	// Get the leader schedule for the current snapshot epoch.
+	leaderSchedulePKIDs, err := utxoView.GetCurrentSnapshotLeaderSchedule()
+	if err != nil {
+		_AddInternalServerError(ww, "GetCurrentEpochProgress: problem fetching current snapshot epoch number")
+		return
+	}
+
+	// Fetch the leader schedule for the current epoch. For each leader in the schedule, we fetch
+	// the public key and username associated with the leader's PKID.
+	leaderSchedule := collections.Transform(leaderSchedulePKIDs, func(pkid *lib.PKID) UserInfoBasic {
+		publicKey := utxoView.GetPublicKeyForPKID(pkid)
+		publicKeyBase58Check := lib.Base58CheckEncode(publicKey, false, fes.Params)
+
+		// Fetch the profile entry for the leader's PKID.
+		profileEntry := utxoView.GetProfileEntryForPKID(pkid)
+		if profileEntry == nil {
+			// If the user has no profile, then we return an empty username.
+			return UserInfoBasic{PublicKeyBase58Check: publicKeyBase58Check, Username: ""}
+		}
+
+		// Happy path: we have both a username and a public key for the leader.
+		return UserInfoBasic{PublicKeyBase58Check: publicKeyBase58Check, Username: string(profileEntry.Username)}
+	})
+
+	// Simulate the current view based on the current tip's view and the amount of time that has elapsed.
+	currentView := currentTip.Header.GetView()
```

Comment:
Should we expose the current view w/ a getter in core? Seems like it would be important to get this right so we have a proper understanding of where this node is.

Reply:
Good idea. I'll make this change.